packages/maskbook/src/plugins/ITO/hooks/usePoolPayload.ts

```diff
@@ -1,26 +1,32 @@
 import { PluginITO_RPC } from '../messages'
 import type { JSON_PayloadInMask } from '../types'
+import { useForceUpdate } from '../../../utils/hooks/useForceUpdate'
 
-const storage = new Map<string, JSON_PayloadInMask>()
-
+const cache = new Map<string, [0, Promise<void>] | [1, JSON_PayloadInMask] | [2, Error]>()
+export function poolPayloadErrorRetry() {
+    cache.forEach(([status], pid) => status === 2 && cache.delete(pid))
+}
 export function usePoolPayload(pid: string) {
-    if (!storage.has(pid)) throw suspender(pid)
-    return {
-        payload: storage.get(pid)!,
-        retry: () => retry(pid),
+    const rec = cache.get(pid)
+    const forceUpdate = useForceUpdate()
+    if (!rec) {
+        const p = suspender(pid)
+            .then((val) => void cache.set(pid, [1, val]))
+            .catch((e) => void cache.set(pid, [2, e]))
+        cache.set(pid, [0, p])
+        throw p
     }
+    if (rec[0] === 1)
+        return {
+            payload: rec[1],
+            retry: () => {
+                if (cache.has(pid)) cache.delete(pid)
+                forceUpdate()
```

Comment:
Please wrap this call in the `startTransition` (see docs at https://reactjs.org/docs/concurrent-mode-reference.html#usetransition).

Reply:
I'm not sure if we should coupling startTransition with retry. In some cases, maybe we want to see the loading state.

We can use startTransition on where we want to use.

https://github.com/DimensionDev/Maskbook/blob/master/packages/maskbook/src/plugins/ITO/UI/ClaimGuide.tsx#L42

Reply:
Fine, if you think it is a necessity to show the loading state then this LGTM

Reply:
okay